src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,190 @@
+        assertThrows(RegistrationException.class, () -> registration.register(user));
+    }
+
+    @Test
+    void when_ageNull_notOk() {
+        user.setAge(null);
+        assertThrows(RegistrationException.class, () -> registration.register(user));
+    }
+
+    @Test
+    void when_userNull_notOk() {
+        user = null;
+        assertThrows(RegistrationException.class, () -> registration.register(user));
+    }
+
+    @Test
+    void when_loginLength6_Ok() {
+        user.setLogin("123456");
+        assertEquals(registration.register(user), user);
+    }
+
+    @Test
+    void when_loginLength5_notOk() {
+        user.setLogin("12345");
+        assertThrows(RegistrationException.class, () -> registration.register(user));
+    }
+
+    @Test
+    void when_loginLength7_Ok() {
+        user.setLogin("1234567");
+        assertEquals(registration.register(user), user);
+    }
+
+    @Test
+    void when_passwordLength7_Ok() {
+        user.setPassword("1234567");
+        assertEquals(registration.register(user), user);
+    }
+
+    @Test
+    void when_passwordLength6_Ok() {
+        user.setPassword("123456");
+        assertEquals(registration.register(user), user);
+    }
+
+    @Test
+    void when_passwordLength5_notOk() {
+        user.setPassword("12345");
+        assertThrows(RegistrationException.class, () -> registration.register(user));
+    }
+
+    @Test
+    void when_loginLength0_notOk() {
+        user.setLogin("");
+        assertThrows(RegistrationException.class, () -> registration.register(user));
+    }
+
+    @Test
+    void when_passwordLength0_notOk() {
+        user.setPassword("");
+        assertThrows(RegistrationException.class, () -> registration.register(user));
+    }
+
+    @Test
+    void when_ageIs18_Ok() {
+        user.setAge(18);
+        assertEquals(registration.register(user), user);
+    }
+
+    @Test
+    void when_ageIsNegative5_notOk() {
+        user.setAge(-5);
+        assertThrows(RegistrationException.class, () -> registration.register(user));
+    }
+
+    @Test
+    void when_ageIs17_notOk() {
+        user.setAge(17);
+        assertThrows(RegistrationException.class, () -> registration.register(user));
+    }
+
+    @Test
+    void when_ageIs19_Ok() {
+        user.setAge(19);
+        assertEquals(registration.register(user), user);
+    }
+
+    @Test
+    void when_ageIs0_notOk() {
+        user.setAge(0);
+        assertThrows(RegistrationException.class, () -> registration.register(user));
+    }
+
+    @Test
+    void check_ifUserAdded_Ok() {
+        registration.register(user);
+        assertEquals(Storage.people.get(0), user);
+    }
+
+    @Test
+    void when_addExistedUser_notOk() {
+        user.setLogin("loginTest");
+        registration.register(user);
+        user = initUser();
+        user.setLogin("loginTest");
+        assertThrows(RegistrationException.class, () -> registration.register(user));
+    }
+
+    @Test
+    void when_add15ValidUsers_Ok() {
```

Comment:
If you already check that valid user can register without errors, you don't need this test case.